### PR TITLE
4 New EC2 Instances with Recommended Configuration

### PR DESCRIPTION
**Pull request by waahm7**
*Description of changes:*
- Adds 2 new P5 and 2 new TRN2 EC2 instance types with recommended configuration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/s3_platform_info.c [hunk 1]
```diff
@@ -68,11 +68,43 @@ static struct aws_s3_platform_info s_p4de_platform_info = {
...
 
+/**** Begin trn2_48_large *****/
+/*
+ * Similar to P5 comment above, we will not be able to get the advertised 3.2Tb bandwidth from CPU.
+ */
+static struct aws_s3_platform_info s_trn2_platform_info = {
+    .instance_type = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("trn2.48xlarge"),
+    /* not all of the advertised 1600 Gbps bandwidth can be hit from the cpu in user-space */
+    .max_throughput_gbps = 400,
+    .has_recommended_configuration = true,
```
Comment (graebm) [inline source/s3_platform_info.c:96]: need to coordinate with python teams on this...

**Review by graebm — COMMENTED** [review]
.

File: source/s3_platform_info.c [hunk 2]
```diff
@@ -44,19 +44,19 @@ static struct aws_s3_platform_info s_c5n_9xlarge_platform_info = {
...
     .instance_type = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("p4de.24xlarge"),
-    .max_throughput_gbps = 400u,
+    .max_throughput_gbps = 400,
     .has_recommended_configuration = true,
 };
 
 /***** End p4d.24xlarge and p4de.24xlarge ****/
 
-/***** Begin p5.48xlarge ******/
+/***** Begin p5_.48xlarge ******/
```
[the remark this answers is not in the document]
Comment (waahm7) [reply source/s3_platform_info.c:59]: It's a section for the p5 family, so `_` means p5+(e/en etc), and `.` is part of the instance name.

[the remark this answers is not in the document]
Comment (waahm7) [reply source/s3_platform_info.c]: I have gotten rid of `_` since it was confusing and made the sections explicit.

**Comment by waahm7** [comment]
We decided not to merge this for now.

**Pull request closed without merging**